app/src/org/commcare/android/view/GraphView.java

```diff
@@ -181,8 +196,26 @@ private void renderSeries(SeriesData s) throws InvalidStateException {
         for (XYPointData d : s.getPoints()) {
             sortedPoints.add(d);
         }
-        Collections.sort(sortedPoints, new PointComparator());
+        Comparator<XYPointData> comparator;
+        if (mData.getType().equals(Graph.TYPE_BAR)) {

```

Comment:
Theoretically, you could specify a bar graph with multiple series that have different nodesets, or the same nodeset but different x functions, or the same nodeset and x function but are sorted differently, and any of those setups will produce misleading data. I'm inclined to not worry about this right now but could put in some error handling if others disagree.